src/components/headless/components/data/item-scroller/transition.component.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+
+interface TransitionProps {
+  children: (state: string) => React.ReactNode;
```

Comment:
I think the return type here should be `TypeOrArray<React.ReactNode>`. Also you could just import `ReactNode` instead of using `React` then you can remove that from the imports.